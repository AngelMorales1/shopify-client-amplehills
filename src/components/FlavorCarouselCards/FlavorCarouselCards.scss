@import 'settings';

.FlavorCarouselCards {
  &__card {
    border-radius: radius('s');
    margin: spacing(4);
    width: spacing(65);

    @media ($medium-up) {
      width: spacing(75);
    }
  }

  &__mark {
    width: spacing(20);
    height: spacing(20);
  }

  &__mark-text {
    width: spacing(20);
    font-size: $font-2;
    letter-spacing: spacing(0.1);
  }

  &__image {
    width: spacing(50);
    height: spacing(50);
    object-fit: contain;
    margin-bottom: spacing(8);

    @media ($medium-up) {
      width: spacing(60);
      height: spacing(60);
      margin-bottom: spacing(12);
    }
  }
}
