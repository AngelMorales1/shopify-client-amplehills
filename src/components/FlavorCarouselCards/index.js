import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import get from 'utils/get';
import cx from 'classnames';

import styles from './FlavorCarouselCards.scss';
import { Image } from 'components/base';

const FlavorCarouselCards = ({ flavorItems, numberOfCardToRender }) => {
  const selectedFlavorItems =
    typeof numberOfCardToRender === 'number'
      ? flavorItems.slice(0, numberOfCardToRender)
      : flavorItems;

  return (
    <Fragment>
      {selectedFlavorItems.map((flavor, i) => {
        const id = get(flavor, '_id', i);
        const title = get(flavor, 'name', '');
        const image = get(flavor, 'image.src', '');
        const label = get(flavor, 'label', '');

        return (
          <div
            key={id}
            className={cx(
              styles['FlavorCarouselCards__card'],
              'bg-white p3 flex flex-column justify-center items-center relative'
            )}
          >
            {label ? (
              <div
                className={cx(
                  styles['FlavorCarouselCards__mark'],
                  'z-subnav circle bg-pink absolute z-1 flex items-center justify-center mr2 mt3 r0 t0 p1'
                )}
              >
                <p
                  className={cx(
                    styles['FlavorCarouselCards__mark-text'],
                    'light center carter text-heavy-gray white-space-normal'
                  )}
                >
                  {label}
                </p>
              </div>
            ) : null}
            <Image
              className={cx(styles['FlavorCarouselCards__image'])}
              src={`${image}?w=300`}
              alt={title ? `${title} image` : ''}
            />
            <p
              className={cx(
                styles['FlavorCarouselCards__title'],
                'bold center pt3'
              )}
            >
              {title}
            </p>
          </div>
        );
      })}
    </Fragment>
  );
};

FlavorCarouselCards.propTypes = {
  flavorItems: PropTypes.array
};

FlavorCarouselCards.defaultProps = {
  flavorItems: []
};

export default FlavorCarouselCards;
