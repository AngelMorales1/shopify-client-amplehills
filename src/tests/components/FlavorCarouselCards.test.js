import React from 'react';
import { shallow } from 'enzyme';

import FlavorCarouselCards from 'components/FlavorCarouselCards';

it('renders without data', () => {
  const component = shallow(<FlavorCarouselCards />);

  expect(component).toMatchSnapshot();
});

it('renders with data', () => {
  const component = shallow(<FlavorCarouselCards flavorItems={[]} />);

  expect(component).toMatchSnapshot();
});
