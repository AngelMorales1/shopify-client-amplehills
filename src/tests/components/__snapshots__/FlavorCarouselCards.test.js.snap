// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with data 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FlavorCarouselCards
    flavorItems={Array []}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [],
    },
    "ref": null,
    "rendered": Array [],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [],
      },
      "ref": null,
      "rendered": Array [],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`renders without data 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FlavorCarouselCards
    flavorItems={Array []}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [],
    },
    "ref": null,
    "rendered": Array [],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [],
      },
      "ref": null,
      "rendered": Array [],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
